src/win/process.c

```diff
@@ -375,6 +423,9 @@ static WCHAR* search_path(const WCHAR *file,
   /* Check if the filename includes an extension */
   dot = wcschr(file_name_start, L'.');
   name_has_ext = (dot != NULL && dot[1] != L'\0');
+  if (name_has_ext) {
+    *isScript = (_wcsicmp(dot, L".bat") == 0 )||(_wcsicmp(dot, L".cmd") == 0);
```

Comment:
@piscisaureus no this code is fine since it is starting from filename component of the original string. However since it serarches for the first occurence of ".", it could get thrown off if you have for example ```c:\some\dir\script.blah.bat```